### PR TITLE
Ogc 1756 translator data check

### PR DESCRIPTION
**Pull request by BreathingFlesh**
Please fill in the commit message below and work through the checklist. You can delete parts that are not needed, e.g. the optional description, the link to a ticket or irrelevant options of the checklist.

## Commit message

Translator: Request translator to check data after a year

TYPE: Feature
LINK: OGC-1756

## Checklist

- [x] I have performed a self-review of my code
- [x] I considered adding a reviewer
- [x] I have updated the PO files
- [x] I have tested my code thoroughly by hand
    - [x] I have tested styling changes/features on different browsers
- [x] I have added tests for my changes/features
    - [x] I am using freezegun for tests depending on date and times

### CODE REVIEW

File: src/onegov/translator_directory/layout.py [hunk 1]
```diff
@@ -186,6 +199,20 @@ def editbar_links(self) -> list[Link | LinkGroup] | None:
...
             ]
+        elif self.translator_data_outdated():
+            return [
+                Link(
+                    _('Report change'),
+                    self.request.link(self.model, name='report-change'),
+                    attrs={'class': 'report-change'}
+                ),
+                Link(
+                    _('Confirm currnet data'),
```
Comment (Tschuppi81) [inline src/onegov/translator_directory/layout.py:210]: little typo

File: src/onegov/translator_directory/layout.py [hunk 2]
```diff
@@ -97,6 +99,17 @@ def __init__(
             request: TranslatorAppRequest
         ) -> None: ...
 
+    def translator_data_outdated(self) -> bool:
+        if self.request.is_translator:
+            tz = pytz.timezone('Europe/Zurich')
+            year_ago = datetime.now(tz=tz) - timedelta(days=365)
```
Comment (Tschuppi81) [inline src/onegov/translator_directory/layout.py:105]: this is not fully accurate for leap years, functionality wise minor
```
import datetime
import pytz
from dateutil.relativedelta import relativedelta

datetime.datetime.now(tz=tz) - timedelta(days=365)
datetime.datetime(2023, 8, 9, 7, 31, 3, 141404, tzinfo=<DstTzInfo 'Europe/Zurich' CEST+2:00:00 DST>)
datetime.datetime.now(tz=tz) - relativedelta(years=1)
datetime.datetime(2023, 8, 8, 7, 31, 33, 475108, tzinfo=<DstTzInfo 'Europe/Zurich' CEST+2:00:00 DST>)
```

File: tests/onegov/translator_directory/test_views.py [hunk 3]
```diff
@@ -72,123 +73,126 @@ def test_view_translator(client):
...
+        # test translator can login and view his own data
+        client.logout()
+        reset_password_url = re.search(
+            r'(http://localhost/auth/reset-password[^)]+)',
+            mail['TextBody']
+        ).group()
+        page = client.get(reset_password_url)
+        page.form['email'] = 'test@test.com'
+        page.form['password'] = 'p@ssw0rd'
+        page.form.submit()
```
Comment (Tschuppi81) [inline tests/onegov/translator_directory/test_views.py:193]:
```suggestion
```suggestion
    with freeze_time('2021-12-31'):
        page = client.login('test@test.com', 'p@ssw0rd', None).maybe_follow()
        assert 'Sind ihre Daten noch aktuell? Bitte überprüfen Sie' not in page
        assert '978654' in page
        assert 'Uncle' in page
        assert 'Bob' in page
```
```

**Review by Tschuppi81 — APPROVED** [review]

File: src/onegov/translator_directory/layout.py (hunk 2 above)
Comment (BreathingFlesh) [reply src/onegov/translator_directory/layout.py:105]: Wouldn't be too problematic in this case, but good to know! I changed it to year now.

**Pull request merged**